app/src/main/java/io/gnosis/safe/ui/settings/owner/ImportOwnerKeyViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package io.gnosis.safe.ui.settings.owner
+
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import pm.gnosis.mnemonic.Bip39Generator
+import javax.inject.Inject
+
+class ImportOwnerKeyViewModel
+@Inject constructor(
+    private val bip39Generator: Bip39Generator,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<ImportOwnerKeyState>(appDispatchers) {
+
+    override fun initialState(): ImportOwnerKeyState = ImportOwnerKeyState.AwaitingInput
+
+    fun validate(seedPhrase: String) {
+        val cleanedUpSeedPhrase = cleanupSeedPhrase(seedPhrase)
+        runCatching { bip39Generator.validateMnemonic(cleanedUpSeedPhrase) }
+            .onFailure { safeLaunch { updateState { ImportOwnerKeyState.Error(ViewAction.ShowError(InvalidSeedPhrase)) } } }
+            .onSuccess { mnemonic ->
+                safeLaunch {
+                    updateState {
+                        if (mnemonic == cleanedUpSeedPhrase) {
+                            ImportOwnerKeyState.ValidSeedPhraseSubmitted(cleanedUpSeedPhrase)
+                        } else {
+                            ImportOwnerKeyState.Error(ViewAction.ShowError(InvalidSeedPhrase))
+                        }
+                    }
+                }
+            }
+    }
+
+    private fun cleanupSeedPhrase(seedPhrase: String): String {
+        return seedPhrase.split("\\s+?|\\p{Punct}+?".toRegex())
```

Comment:
This is a nice addition, but please communicate this also to liliya and the ios team, so that it can be tested and is alligned between different clients. The specs say different things at different points.

Docs: `Before performing validation, strip whitespaces and make text lowercase (not visible to a user)`
Android issue: `Strip or ignore extra whitespace and punctuation symbols in the phrase.`